ironfish/src/network/messages/getCompactBlock.test.ts

```diff
@@ -18,28 +20,52 @@ describe('GetCompactBlockRequest', () => {
 })
 
 describe('GetCompactBlockResponse', () => {
-  it('serializes the object into a buffer and deserializes to the original object', () => {
-    const compactBlock: SerializedCompactBlock = {
-      header: {
-        graffiti: Buffer.alloc(32, 'graffiti1', 'utf8').toString('hex'),
-        minersFee: '0',
-        noteCommitment: {
+  const nodeTest = createNodeTest()
+
+  function expectGetCompactBlockResponseToMatch(
+    a: GetCompactBlockResponse,
+    b: GetCompactBlockResponse,
+  ): void {
+    // Test transactions separately because Transaction is not a primitive type
+    expect(a.compactBlock.transactions.length).toEqual(b.compactBlock.transactions.length)
+    a.compactBlock.transactions.forEach((transactionA, transactionIndexA) => {
+      const transactionB = b.compactBlock.transactions[transactionIndexA]
+
+      expect(transactionA.index).toEqual(transactionB.index)
+      expect(transactionA.transaction.hash().equals(transactionB.transaction.hash())).toBe(true)
+    })
+
+    expect({
+      ...a,
+      compactBlock: { ...a.compactBlock, transactions: undefined },
+    }).toMatchObject({ ...b, compactBlock: { ...b.compactBlock, transactions: undefined } })
+  }
+
+  // eslint-disable-next-line jest/expect-expect
+  it('serializes the object into a buffer and deserializes to the original object', async () => {
+    const { transaction } = await useTxSpendsFixture(nodeTest.node)
+
+    const compactBlock: CompactBlock = {
+      header: new BlockHeader(
+        2,
+        Buffer.alloc(32, 2),
+        {
           commitment: Buffer.alloc(32, 1),
           size: 1,
         },
-        nullifierCommitment: {
-          commitment: Buffer.alloc(32, 2).toString('hex'),
+        {
+          commitment: Buffer.alloc(32, 2),
           size: 2,
         },
-        previousBlockHash: Buffer.alloc(32, 2).toString('hex'),
-        randomness: '1',
-        sequence: 2,
-        target: '12',
-        timestamp: 200000,
-      },
+        new Target(12),
+        BigInt(1),
+        new Date(200000),
+        BigInt(0),
+        Buffer.alloc(32, 'graffiti1', 'utf8'),
+      ),
       transactions: [
-        { transaction: Buffer.from('foo'), index: 0 },
-        { transaction: Buffer.from('bar'), index: 2 },
+        { transaction: transaction, index: 0 },
+        { transaction: transaction, index: 2 },
```

Comment:
Probably worth having two unique transactions here as well

Reply:
Changed to having 2 unique transactions